community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/commands/predicates/Predicate.scala

```diff
@@ -167,52 +195,64 @@ case class PropertyExists(variable: Expression, propertyKey: KeyToken) extends P
 
 trait StringOperator {
   self: Predicate =>
+
   override def isMatch(m: ExecutionContext)(implicit state: QueryState) = (lhs(m), rhs(m)) match {
-    case (l: String, r: String) => Some(compare(l,r))
+    case (l: TextValue, r: TextValue) => Some(compare(l.stringValue(), r.stringValue()))
     case (_, _) => None
   }
 
   def lhs: Expression
+
   def rhs: Expression
+
   def compare(a: String, b: String): Boolean
+
   override def containsIsNull = false
+
   override def arguments = Seq(lhs, rhs)
+
   override def symbolTableDependencies = lhs.symbolTableDependencies ++ rhs.symbolTableDependencies
 }
 
 case class StartsWith(lhs: Expression, rhs: Expression) extends Predicate with StringOperator {
+
   override def compare(a: String, b: String) = a.startsWith(b)
 
   override def rewrite(f: (Expression) => Expression) = f(copy(lhs.rewrite(f), rhs.rewrite(f)))
 }
 
 case class EndsWith(lhs: Expression, rhs: Expression) extends Predicate with StringOperator {
+
   override def compare(a: String, b: String) = a.endsWith(b)
 
   override def rewrite(f: (Expression) => Expression) = f(copy(lhs.rewrite(f), rhs.rewrite(f)))
 }
 
 case class Contains(lhs: Expression, rhs: Expression) extends Predicate with StringOperator {
+
   override def compare(a: String, b: String) = a.contains(b)
 
   override def rewrite(f: (Expression) => Expression) = f(copy(lhs.rewrite(f), rhs.rewrite(f)))
 }
 
-case class LiteralRegularExpression(lhsExpr: Expression, regexExpr: Literal)(implicit converter: String => String = identity) extends Predicate {
-  lazy val pattern = converter(regexExpr.v.asInstanceOf[String]).r.pattern
+case class LiteralRegularExpression(lhsExpr: Expression, regexExpr: Literal)
+                                   (implicit converter: TextValue => TextValue = identity) extends Predicate {
+
+  lazy val pattern = converter(regexExpr.anyVal.asInstanceOf[TextValue]).stringValue().r.pattern
 
   def isMatch(m: ExecutionContext)(implicit state: QueryState) =
     lhsExpr(m) match {
-      case s: String => Some(pattern.matcher(s).matches())
+      case s: TextValue => Some(pattern.matcher(s.stringValue()).matches())
       case _ => None
     }
 
   def containsIsNull = false
 
   def rewrite(f: (Expression) => Expression) = f(regexExpr.rewrite(f) match {
-    case lit: Literal => LiteralRegularExpression(lhsExpr.rewrite(f), lit)(converter)
-    case other        => RegularExpression(lhsExpr.rewrite(f), other)(converter)
-  })
+                                                   case lit: Literal => LiteralRegularExpression(lhsExpr.rewrite(f),
```

Comment:
This file is full of reformating changes, some of which (like this one) are quite ugly. Did we need to do this much reformatting?